Katello/katello
# Fixes 36726 - Non-admin users with the view_smart_proxies permission …

…can now access capsules from the API.

#### What are the changes introduced in this pull request?
Non admin users before this change could not access capsules using the `/katello/api/capsules/<id>` endpoint. This has been fixed and now requires the user to have the `view_smart_proxies` permission.

#### Considerations taken when implementing this change?
Users with this permission could previously see capsule content using the `/katello/api/capsules/` endpoint. This change brings the other endpoint into parity.

#### What are the testing steps for this pull request?

1. Launch a Katello instance with at least one capsule.
2. Create two non-admin users. Make a new role with the `view_smart_proxies` permission. Add this role to one of the two new users.
3. Check the `/capsules` endpoint with the admin user and the two new users. The user without the view permission should receive an error, while the other two should list the contents of all capsules.
```
curl -k -u <user>:<password> <address>/katello/api/capsules/
```
4. Check the `/capsules/<id>` endpoint for a specific capsule. The user without the view permission should receive an error, while the other two should list the contents of the capsule.
```
curl -k -u <user>:<password> <address>/katello/api/capsules/<id>
```
